### ISSUE TITLE
Block (or redirect) universally 1x1 gif images

### ISSUE THREAD

**Issue by peace2000** [uBlockOrigin/uBlock-issues#1705]
### Prerequisites

- [X] I verified that this is not a filter issue (MUST be reported at [filter issue tracker](https://github.com/uBlockOrigin/uAssets/issues))
- [X] This is not a [support issue or a question](https://old.reddit.com/r/uBlockOrigin/)
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uBlock-issues/issues) to avoid opening a duplicate issue
- [X] The issue is not present after wholly disabling uBlock Origin ("uBO") in the browser
- [X] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I report is not a normal behavior

### I tried to reproduce the issue when...

- [X] uBO is the only extension
- [X] uBO with default lists/settings
- [X] using a new, unmodified browser profile

### Description

Would it be possible to implement a feature (or a built-in function) to uBO to block or redirect universally all 1x1 gifs as the main purpose of those is to track user? Websites name them differently so it's not possible to use one rule to block or redirect them all and there's a constant need for filter list maintainers to add more rules to block or redirect them on different websites.

If something like this were to be implemented, I think it'd safest to use redirection rather than blocking as sometimes blocking those gifs will cause issues.

Examples of sites where redirection is needed instead of blocking to prevent issues:

If you block 1x1 gif on `https://mognavi.jp/`, some content on the front page wont load.
`||mognavi.jp/image/1x1.gif`

But, if you redirect it, everything loads properly:
`||mognavi.jp/image/1x1.gif$redirect=1x1.gif`

Another sample, this page `https://news.ycombinator.com/item?id=28306390` uses 1x1 gif for content alignment purposes, if you block it, content alignment will look different:
`||news.ycombinator.com/s.gif`

But, if you redirect it, alignment works properly:
`||news.ycombinator.com/s.gif$redirect=1x1.gif`

Next sample page (`rakentaja.fi`) has a tracking pixel that's not covered by filter lists because of it's non-generic name, but blocking of it won't cause issues.

### A specific URL where the issue occurs

`https://www.rakentaja.fi/artikkelit/16202/turvalliset_ja_tehokkaat_eristeet_rakentamiseen_ja_remontointiin.htm`

### Steps to Reproduce

1. Open the page (without any active explicit rule to block `pi.gif`)

### Expected behavior

uBO would block or redirect that 1x1 gif even without explicit blocking or redirection rule.

### Actual behavior

uBO won't block or redirect it as `pi.gif` is not currently found on any known privacy related list (at least not on any uBO's stock lists).

### uBlock Origin version

1.37.3b21

### Browser name and version

Firefox 91.0.2

### Operating System and version

Windows 10

**Comment by peace2000** [comment]
I don't consider this as an duplicate of https://github.com/uBlockOrigin/uBlock-issues/issues/92 because it was slightly different and unclear.

**Comment by uBlock-user** [comment]
> Would it be possible to implement a feature (or built-in function) to uBO to block or redirect universally all 1x1 gifs 

Would require **fetching** the size of all unblocked images and a run a test to see if they're of  `1x1`  size...

**Comment by peace2000** [comment]
> 
> 
> > Would it be possible to implement a feature (or built-in function) to uBO to block or redirect universally all 1x1 gifs
> 
> Would require **fetching** the size of all unblocked images and a run a test to see if they're of `1x1` size...

All right so this will probably be declined?

**Comment by uBlock-user** [comment]
upto to @gorhill

**Comment by gwarser** [comment]
Learning approach will make it fingerprintable, similar to Privacy Badger.

**Comment by gorhill** [comment]
For all the cases provided, the pixel image is _not_ blocked with default filter lists, so there is no point worrying about page breakage.

For none of the cases provided is the case made that there is "tracking": the pixel images are first-party and are devoid of any user data beyond what is expected when downloading a resource, as is also the case for all other 1st-party resources -- so I don't see the point of worrying about these cases in the first place.

Additionally, as @uBlock-user pointed out, it's not possible to know the size of an image before it is downloaded - hence why the [documentation states](https://github.com/gorhill/uBlock/wiki/Per-site-switches#no-large-media-elements) that the _"Block media elements larger than ..."_ has no privacy value, because the resource must be fetched in order to look at the response headers:

> Note that this feature has no privacy value: a connection to the remote server must be performed in order to fetch the size of the resource.

I will decline because the issue is just asking for a (impossible) solution to an unfounded issue as per cases brought forth.

**Comment by peace2000** [comment]
The reason I mentioned those two pages was to point out that 1x1 gif blocking universally would cause issues and that if this were to be implemented, an universal redirection would be safer than universal blocking.

But anyway, as this it not possible there is no point to continue with this. Thanks anyway for your time @gorhill

**Comment by v00v** [comment]
`user pointed out, it's not possible to know the size of an image before it is downloaded`

**NOT TRUE**

Example Using CURL:
``curl -s https://github.githubassets.com/images/modules/logos_page/Octocat.png | identify -format "%wx%h" -``

Result: ``800x665``

**Comment by gorhill** [comment]
> NOT TRUE

Your curl command downloads the image from the server.
